### PR TITLE
Add initialize phase when call sonar:sonar

### PR DESCRIPTION
**Pull request by avdev4j**
Fix the test-integrations scripts for sonar analysis.
With the properties maven plugin we need to have, at least, an initialize phase to load sonar-project.properties.

related to #9423

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by PierreBesson** [comment]
i think you also need to update the README

File: generators/common/templates/README.md.ejs [hunk 1]
```diff
@@ -289,6 +289,11 @@ Then, run a Sonar analysis:
 ```
 ./mvnw -Pprod clean verify sonar:sonar
 ```
+If you need to re-run the sonar phase, please be sur to specify at least the `initialize` phase since sonar properties are loaded from the sonar-project.properties file.
```
Comment (DanielFran) [inline generators/common/templates/README.md.ejs]: be *sure

Comment (avdev4j) [reply generators/common/templates/README.md.ejs]: 👍

File: generators/common/templates/README.md.ejs [hunk 2]
```diff
@@ -280,7 +280,7 @@ Sonar is used to analyse code quality. You can start a local Sonar server (acces
 docker-compose -f src/main/docker/sonar.yml up -d
 ```
 
-You can both run sonar with maven or gradle but also with a sonar-scanner.
+You can both run Sonar with maven or gradle but also with a sonar-scanner.
```
Comment (PierreBesson) [inline generators/common/templates/README.md.ejs]: I would rephrase the sonar scanner sentence. It is a bit annoying as when using --skip-client you are using neither maven or gradle so it makes no sense to mention them ! So I would write it like that:

`You can run a Sonar analysis with using the [sonar-scanner](https://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner)<% if (buildTool) { %> or by using the <% if (buildTool === 'maven') { %>maven<% } %><% if (buildTool === 'gradle') { %>gradle<% } %> plugin<% } %> .`

Comment (avdev4j) [reply generators/common/templates/README.md.ejs]: OFC it's better and I alslo remove the outdated sonar-scanner part below

**Comment by avdev4j** [comment]
close/reopen for CI

**Comment by jdubois** [comment]
@avdev4j can you check also https://www.jhipster.tech/code-quality/ as I think the documentation there will need to be updated accordingly

**Comment by jdubois** [comment]
@avdev4j when testing locally I get:

```
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project sample: Error during parsing of generic test execution report '/Users/julien/workspace/sample/target/test-results/jest/TESTS-results-sonar.xml'. Look at the SonarQube documentation to know the expected XML format.: /Users/julien/workspace/sample/target/test-results/jest/TESTS-results-sonar.xml (No such file or directory)
```

So I'm not sure this is all OK

**Comment by avdev4j** [comment]
@jdubois 
already done here https://github.com/jhipster/jhipster.github.io/pull/714

But I wonder why we need to specify a sonar URL `./mvnw -Pprod clean test sonar:sonar -Dsonar.host.url=http://localhost:9001`since it has the same value than the one in the sonar-project.properties

Also we need to merge this https://github.com/jhipster/jhipster.github.io/pull/706 to add integration tests in the Sonar analysis.
Of course this will only work with JHipster 6+

**Comment by avdev4j** [comment]
> @avdev4j when testing locally I get:
> 
> ```
> [ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project sample: Error during parsing of generic test execution report '/Users/julien/workspace/sample/target/test-results/jest/TESTS-results-sonar.xml'. Look at the SonarQube documentation to know the expected XML format.: /Users/julien/workspace/sample/target/test-results/jest/TESTS-results-sonar.xml (No such file or directory)
> ```
> So I'm not sure this is all OK

Which maven command have you used ?

**Comment by jdubois** [comment]
@avdev4j I did:

`./mvnw clean verify` to run my tests, and then `./mvnw initialize org.jacoco:jacoco-maven-plugin:prepare-agent sonar:sonar \
        -Dsonar.host.url=http://localhost:9001 \                -Dsonar.host.url=http://localhost:9001`

**Comment by avdev4j** [comment]
oooooohh
I found it !

You need to add `-Pprod` to maven command because `webpack:test` script is only run on prod profile in the pom.xml file
